eventsourcing/src/main/java/org/axonframework/eventsourcing/PayloadConvertingEntityEvolver.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.function.BiFunction;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * An {@link EntityEvolver} implementation that converts the {@link EventMessage#getPayload()} to the given
+ * {@code payloadType} to evolve an entity with.
+ * <p>
+ * Will throw a {@link ClassCastException} if the {@code payloadType} does not match.
+ *
+ * @param <P> The payload type of the event to apply.
+ * @param <E> The entity type to evolve.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class PayloadConvertingEntityEvolver<P, E> implements EntityEvolver<E> {
```

Comment:
Why not `PayloadBasedEntityEvolver`? We already have `PayloadBasedTagResolver`.

Reply:
I like it! I'll rename it. :+1: